Source/WebKit/gtk/tutorial.md

```diff
@@ -0,0 +1,382 @@
+
+We would like to add these embellishments to the window to be created, hence we’ll add a function `create_window(GtkApplication *app, GtkWidget *web_view, const char *url)` to create our custom window and call it in `on_app_activate()`.
+
+### Entry box
+
+We will add an input entry box to display URL, the reload button, and the “Stop Loading” button. Then stack it vertically on top of the view of the webpage.
+
+```c
+    GtkWidget *vbox = gtk_box_new(GTK_ORIENTATION_VERTICAL, 0);
+
+    GtkWidget *entry = g_object_new(GTK_TYPE_ENTRY,
+                                    "enable-emoji-completion", FALSE,
+                                    "input-purpose", GTK_INPUT_PURPOSE_URL,
+                                    "primary-icon-name", "view-refresh-symbolic",
+                                    "primary-icon-activatable", TRUE,
+                                    "primary-icon-sensitive", TRUE,
+                                    "primary-icon-tooltip-markup", "Reload",
+                                    "secondary-icon-name", "process-stop-symbolic",
+                                    "secondary-icon-activatable", TRUE,
+                                    "secondary-icon-tooltip-markup", "Stop Loading",
+                                    NULL);
+    gtk_box_pack_start(GTK_BOX(vbox), entry, FALSE, FALSE, 0);
+    gtk_box_pack_end(GTK_BOX(vbox), web_view, TRUE, TRUE, 0);
+```
+We then would like to propagate URL changes to the URL entry as the user navigates and page title changes to the title of the window containing the web view, so we bind the values to the corresponding components:
+
+```c
+    g_object_bind_property(web_view, "uri", entry, "text", G_BINDING_DEFAULT);
+    g_object_bind_property(web_view, "title", window, "title", G_BINDING_DEFAULT);
+```
+
+### Enable buttons’ functionality
+
+We should allow users to click on the reload buttons as well as to see the webpage loaded after finishing typing in the entry box and hitting “Enter” key, by having two callback functions:
+
+```c
+static void
+on_url_entry_activate(GtkEntry *entry, WebKitWebView *web_view)
+{
+    webkit_web_view_load_uri(web_view, gtk_entry_get_text(entry));
+    gtk_widget_grab_focus(GTK_WIDGET(web_view));
+}
+
+static void
+on_url_entry_icon_released(GtkEntry *entry,
+                           GtkEntryIconPosition icon,
+                           GdkEvent *event,
+                           WebKitWebView *web_view)
+{
+    switch (icon) {
+        case GTK_ENTRY_ICON_PRIMARY:
+            webkit_web_view_reload(web_view);
+            break;
+        case GTK_ENTRY_ICON_SECONDARY:
+            webkit_web_view_stop_loading(web_view);
+            break;
+    }
+}
+```
+
+Then register the two functions for the components in `create_window()`:
```

Comment:
Better replace “register” with “connect”, for consistency with the usual wording for GObject signals.